File: ros_basics/scripts/msg_publisher.py
#!/usr/bin/env python
import rospy 
from std_msgs.msg import String, Int32
from ros_basics.msg import greet

def publisher():
	rospy.init_node("msg_publisher", anonymous=True)
	pub = rospy.Publisher('greet', greet, queue_size=100)

	rate = rospy.Rate(10)

	while not rospy.is_shutdown():

File: ros_basics/src/msg_publisher.cpp
#include "ros/ros.h"
#include "std_msgs/Int32.h"
#include "ros_basics/greet.h"
#include <iostream>

int main(int argc, char **argv)
{
	ros::init(argc, argv, "msg_publisher");
	ros::NodeHandle nh;
	ros::Publisher pub = nh.advertise<ros_basics::greet>("/greeting",10);
	ros::Rate loop_rate(10);

	int num = 0;

	while(ros::ok())
	{
		ros_basics::greet msg;
		std::stringstream ss;
		ss << "Hello ROS user!" ;
		msg.greeting = ss.str();

		msg.number = num;

		ROS_INFO("Greeting: %s", msg.greeting.c_str());
		ROS_INFO("Number: %d", msg.number);
		pub.publish(msg);
		ros::spinOnce();
		loop_rate.sleep();
		++num;
	}
	return 0;
}